src/main/java/hudson/plugins/git/extensions/GitSCMChangelogExtension.java

```diff
@@ -0,0 +1,33 @@
+package hudson.plugins.git.extensions;
+
+import java.io.IOException;
+
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitException;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.Revision;
+import org.jenkinsci.plugins.gitclient.ChangelogCommand;
+import org.jenkinsci.plugins.gitclient.GitClient;
+
+/**
+ * FIXME JavaDoc
+ * @author Zhenlei Huang
+ */
+public abstract class GitSCMChangelogExtension extends FakeGitSCMExtension {
```

Comment:
For example, this is something I don't understand. Why extending `FakeGitSCMExtension` instead of `GitSCMExtension`

Reply:
I’m not sure which one is better. If I remember correctly, extending GitSCMExtension is also OK.